src/api/courses.js

```diff
@@ -254,15 +257,32 @@ router.patch(
   ],
   safeAsync(async (req, res, _next) => {
     const courseId = res.locals.course.dataValues.id
-    const { course } = res.locals
+    const {
+      locals: { userAuthz, course },
+    } = res
     const data = matchedData(req)
-    await course.update({
-      name: data.name !== null ? data.name : undefined,
-      shortcode: data.shortcode !== null ? data.shortcode : undefined,
-      isUnlisted: data.isUnlisted !== null ? data.isUnlisted : undefined,
-      questionFeedback:
-        data.questionFeedback !== null ? data.questionFeedback : undefined,
-    })
+
+    if (userAuthz.isAdmin) {
+      await course.update({
+        name: data.name !== null ? data.name : undefined,
+        shortcode: data.shortcode !== null ? data.shortcode : undefined,
+        isUnlisted: data.isUnlisted !== null ? data.isUnlisted : undefined,
+        questionFeedback:
+          data.questionFeedback !== null ? data.questionFeedback : undefined,
+      })
+    } else {
+      if (data.name !== course.name || data.shortcode !== course.shortcode) {
```

Comment:
I believe this will erroneously cause an error if the `PATCH` request doesn't contain `name` or `shortcode`, which should be valid - I should be able to independently patch any part of an object without sending back the current state. Can you write some tests to ensure that we have that behavior (e.g. that I can just `PATCH { isUnlisted: false }` without needing to include a name or shortcode) and update this code if needed?

Reply:
Good point, I forgot that I made it mandatory to pass the name and shortcode in. I'll write a few test cases to make sure this doesn't fail